src/classes/SRVC_ContactRelationshipAutoCreate.cls

```diff
@@ -0,0 +1,175 @@
+
+        for (MODL_ConRelationshipAutoCreateMapping conRelAutoCreateMappingMdl : contactRelationShipAutoCreateMapModelsList){
+            Object newContactFieldValue = newContact.get(conRelAutoCreateMappingMdl.contactFieldApiName);
+            Object oldContactFieldValue = oldContact.get(conRelAutoCreateMappingMdl.contactFieldApiName);
+    
+            if (newContactFieldValue != null && newContactFieldValue != oldContactFieldValue){
+                Relationship__c relationshipToAdd = new Relationship__c(Contact__c = newContact.Id,
+                                                                        RelatedContact__c = (Id)newContactFieldValue,
+                                                                        Type__c = conRelAutoCreateMappingMdl.relationshipType,
+                                                                        Status__c = 'Current');
+
+                contactRelationshipsModel.addRelationship(relationshipToAdd);
+            }
+        }
+
+        return contactRelationshipsModel;
+    }
+
+    /**************************************************************************************************************************
+    * @description Retrieves the Relationship records to insert from a specified ContactsRelationships model.
+    * @param contactRelationshipsModel The MODL_ContactsRelationships model from which to retrieve Relationship records to insert.
+    * @return A list of Relationship records to be inserted.
+    **************************************************************************************************************************/
+    public virtual List<Relationship__c> getRelationshipsToInsertFromModel(MODL_ContactsRelationships contactsRelationshipsModel){
+        if (UTIL_CustomSettingsFacade.getSettings().Allow_AutoCreated_Duplicates__c == true){
+            return contactsRelationshipsModel.relationshipList;
+        }
+
+        List<Relationship__c> contactRelationships = [SELECT Id,
+                                                             Contact__c,
+                                                             RelatedContact__c,
+                                                             Type__c
+                                                      FROM Relationship__c
+                                                      WHERE Contact__c in :contactsRelationshipsModel.contactIdsSet
+                                                      AND RelatedContact__c in :contactsRelationshipsModel.relatedContactIdSet
+                                                      LIMIT 2000];
+        
+        for (Relationship__c rel : contactRelationships){
+            Relationship__c relToRemove = this.getDuplicateRelationship(rel, contactsRelationshipsModel);
+            if (relToRemove != null){
+                contactsRelationshipsModel.removeRelationshipFromList(relToRemove);
+            } 
+        }
```

Comment:
Given all the guff this is giving us, let's pull down contactsRelationshipsModel.relationshipList first and make a set from it. We can then perform set.remove(relToRemove). We have to cast it back to a list to return it, but this also gives us a side-effect free way to make a new collection.  Were we not dealing with the DMLWrapper downstream, I would simply change this method signature to a Set